sphinx/themes/basic/static/searchtools.js

```diff
@@ -58,6 +58,15 @@ const _removeChildren = (element) => {
 const _escapeRegExp = (string) =>
   string.replace(/[.*+\-?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
 
+const _escapeHTML = (text) => {
+  return text
+    .replaceAll("&", "&amp;")
+    .replaceAll("<", "&lt;")
+    .replaceAll(">", "&gt;")
+    .replaceAll('"', "&quot;")
+    .replaceAll("'", "&#39;");
```

Comment:
I think that since we produce valid XML (ref: #12168) we may be able to use `&apos;` here and drag ourselves into the 1990s:

```suggestion
    .replaceAll("'", "&apos;");
```